### PR TITLE
Add port to task list on query detail page

### PR DESCRIPTION
**Pull request by raghavsethi**
Required reducing the font size, and converting some headers to icons. Supersedes #5513.

### CODE REVIEW

**Comment by raghavsethi** [comment]
@electrum made the ports only show when there were multiple workers on the same host.

**Comment by raghavsethi** [comment]
ping @electrum

File: presto-main/src/main/resources/webapp/assets/utils.js [hunk 1]
```diff
@@ -33,6 +33,10 @@ var QUERY_STATE_COLOR_MAP = {
 
 function getReadableErrorCode(errorType, errorCode)
 {
+    if (typeof someVar === 'undefined') {
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/utils.js]: `someVar`?

File: presto-main/src/main/resources/webapp/assets/utils.js [hunk 2]
```diff
@@ -145,6 +145,15 @@ function getHostname(url) {
     return hostname;
 }
 
+function getPort(url) {
+    return new URL(url).port;
+}
+
+function getHostWithPort(url) {
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/utils.js]: I'd name this `getHostAndPort`

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 3]
```diff
@@ -36,13 +36,31 @@ var TaskList = React.createClass({
...
         return 0;
     },
+    showPortNumbers: function(tasks) {
+        if (tasks.length <= 1) {
+            return false;
+        }
+
+        var hostToPortNumber = {};
+        for (var i = 0; i < tasks.length; i++) {
+            if ((getHostname(tasks[i].taskStatus.self) in hostToPortNumber) && (hostToPortNumber[getHostname(tasks[i].taskStatus.self)] != getPort(tasks[i].taskStatus.self))) {
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query.js]: Line break after `&&`

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 4]
```diff
@@ -36,13 +36,31 @@ var TaskList = React.createClass({
...
+                return true;
+            }
+            hostToPortNumber[getHostname(tasks[i].taskStatus.self)] = getPort(tasks[i].taskStatus.self);
+        }
+
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query.js]:
```js
var hostToPortNumber = {};
for (var i = 0; i < tasks.length; i++) {
    var task = tasks[i].taskStatus.self;
    var hostname = getHostname(task);
    if ((hostname in hostToPortNumber) && (hostToPortNumber[hostname] != getPort(task))) {
        return true;
    }
    hostToPortNumber[hostname] = getPort(task);
}

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 5]
```diff
@@ -36,13 +36,31 @@ var TaskList = React.createClass({
 
         return 0;
     },
+    showPortNumbers: function(tasks) {
+        if (tasks.length <= 1) {
```
Comment (electrum) [inline presto-main/src/main/resources/webapp/assets/query.js]: This special case isn't need

**Pull request closed without merging**